Python/gc.c

```diff
@@ -851,9 +858,8 @@ handle_weakrefs(PyGC_Head *unreachable, PyGC_Head *old)
 
             /* Move wr to wrcb_to_call, for the next pass. */
             wrasgc = AS_GC((PyObject *)wr);
-            assert(wrasgc != next); /* wrasgc is reachable, but
-                                       next isn't, so they can't
-                                       be the same */
+            // wrasgc is reachable, but next isn't, so they can't be the same
+            _PyObject_ASSERT(wr, wrasgc != next);
```

Comment:
There's a GitHub actions warning: I think this needs a `(PyObject *)` cast on `wr`.

Reply:
Alright, I missed that warning. It's now fixed.